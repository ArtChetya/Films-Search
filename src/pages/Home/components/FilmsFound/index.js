import React from 'react';
import PropTypes from 'prop-types';
import grey from '@material-ui/core/colors/grey';
import { ColorText } from '../../../../components/ColorText';

export const FilmsFound = ({ films }) => {
    return (
        <ColorText variant="subtitle2" fontcolor={grey[900]}>
            {films.length} movies found
        </ColorText>
    );
};

FilmsFound.propTypes = {
    films: PropTypes.array.isRequired
};
