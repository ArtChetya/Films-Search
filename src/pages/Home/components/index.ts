export * from './FilmsFound';
export * from './FilterBy';
export * from './SearchBy';
export * from './SearchForm';
export * from './SortBy';
