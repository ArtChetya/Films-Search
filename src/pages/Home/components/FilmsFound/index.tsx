import grey from '@material-ui/core/colors/grey';
import { ColorText } from 'components';
import { IFilmDetails } from 'features/filmDetails';
import React, { FunctionComponent } from 'react';

interface IFilmsFound {
    films: IFilmDetails[];
}

export const FilmsFound: FunctionComponent<IFilmsFound> = ({ films }) => {
    return (
        <ColorText variant="subtitle2" fontcolor={grey[900]}>
            {films.length} movies found
        </ColorText>
    );
};
