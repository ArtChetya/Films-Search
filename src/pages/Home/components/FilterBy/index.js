import React from 'react';
import PropTypes from 'prop-types';
import Grid from '@material-ui/core/Grid';
import { ColorButton } from '../../../../components/ColorButton';
import { TransformText } from '../../../../components/TransformText';

export const FilterBy = ({
    config,
    options,
    selectedOptionId,
    onOptionChange
}) => {
    return (
        <Grid container alignItems="center" spacing={8}>
            <Grid item>
                <TransformText
                    fontcolor={config.defaultColor}
                    transform={config.titleTextTransform}
                    variant={config.titleVariant}
                >
                    {config.title}
                </TransformText>
            </Grid>

            {options.map(({ id, label }) => (
                <Grid key={id} item>
                    <ColorButton
                        onClick={() => onOptionChange(id)}
                        variant={config.buttonVariant}
                        backgroundcolor={
                            id === selectedOptionId
                                ? config.activeBgColor
                                : config.defaultBgColor
                        }
                    >
                        <TransformText
                            fontcolor={
                                id === selectedOptionId
                                    ? config.activeColor
                                    : config.defaultColor
                            }
                            transform={config.buttonTextTransform}
                            variant={config.titleVariant}
                        >
                            {label}
                        </TransformText>
                    </ColorButton>
                </Grid>
            ))}
        </Grid>
    );
};

FilterBy.propTypes = {
    config: PropTypes.object.isRequired,
    options: PropTypes.array.isRequired,
    selectedOptionId: PropTypes.string.isRequired,
    onOptionChange: PropTypes.func.isRequired
};
