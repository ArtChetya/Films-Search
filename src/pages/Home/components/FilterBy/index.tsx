import Grid from '@material-ui/core/Grid';
import { ColorButton, TransformText } from 'components';
import React, { FunctionComponent } from 'react';
import { IFilterConfig, IOption } from '../../types';

interface IFilterByProps {
    config: IFilterConfig;
    options: IOption[];
    selectedOptionId: string;
    onOptionChange: (id: string) => void;
}

export const FilterBy: FunctionComponent<IFilterByProps> = ({
    config,
    options,
    selectedOptionId,
    onOptionChange
}) => {
    return (
        <Grid container alignItems="center" spacing={8}>
            <Grid item>
                <TransformText
                    fontcolor={config.defaultColor}
                    transform={config.titleTextTransform}
                    variant={config.titleVariant}
                >
                    {config.title}
                </TransformText>
            </Grid>

            {options.map(({ id, label }) => (
                <Grid key={id} item>
                    <ColorButton
                        onClick={() => onOptionChange(id)}
                        variant={config.buttonVariant}
                        backgroundcolor={
                            id === selectedOptionId
                                ? config.activeBgColor
                                : config.defaultBgColor
                        }
                    >
                        <TransformText
                            fontcolor={
                                id === selectedOptionId
                                    ? config.activeColor
                                    : config.defaultColor
                            }
                            transform={config.buttonTextTransform}
                            variant={config.titleVariant}
                        >
                            {label}
                        </TransformText>
                    </ColorButton>
                </Grid>
            ))}
        </Grid>
    );
};
