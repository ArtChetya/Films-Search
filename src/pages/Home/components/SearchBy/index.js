import React from 'react';
import PropTypes from 'prop-types';
import pink from '@material-ui/core/colors/pink';
import grey from '@material-ui/core/colors/grey';
import Grid from '@material-ui/core/Grid';
import { ColorButton } from '../../../../components/ColorButton';
import { UppercasedText } from '../../../../components/UppercasedText';

export function SearchBy({ searchByLabels, selectedLabel, setSelectedLabel }) {
    return (
        <Grid container alignItems="center" spacing={8}>
            <Grid item>
                <UppercasedText variant="body1">Search By</UppercasedText>
            </Grid>

            {searchByLabels.map(label => (
                <Grid key={label} item>
                    <ColorButton
                        onClick={() => setSelectedLabel(label)}
                        fontcolor="#fff"
                        backgroundcolor={
                            label === selectedLabel ? pink[400] : grey[700]
                        }
                    >
                        {label}
                    </ColorButton>
                </Grid>
            ))}
        </Grid>
    );
}

SearchBy.propTypes = {
    searchByLabels: PropTypes.array.isRequired,
    selectedLabel: PropTypes.string.isRequired,
    setSelectedLabel: PropTypes.func.isRequired
};
