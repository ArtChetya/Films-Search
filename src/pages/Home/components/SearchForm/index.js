import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import pink from '@material-ui/core/colors/pink';
import Grid from '@material-ui/core/Grid';
import TextField from '@material-ui/core/TextField';
import { ColorButton, TransformText } from 'components';
import { SearchBy } from '..';

const SearchField = styled(TextField)`
    && {
        margin: 8px 0;

        div:after {
            border-bottom: 2px solid ${pink[400]};
        }

        & input {
            color: #fff;
        }
    }
`;

export const SearchForm = ({ onSearch }) => {
    const [searchField, setSearchField] = useState('');

    const searchByOptionsList = [
        { id: 'title', label: 'Title' },
        { id: 'genres', label: 'Genre' }
    ];
    const [searchByOptions] = useState(searchByOptionsList);
    const [searchBy, setSearchBy] = useState(searchByOptionsList[0].id);

    const onSubmit = event => {
        event.preventDefault();
        onSearch(searchField, searchBy);
    };

    return (
        <form onSubmit={onSubmit}>
            <Grid container justify="space-between" alignItems="center">
                <Grid item xs={12}>
                    <TransformText transform="uppercase" variant="h6">
                        find your movie
                    </TransformText>
                </Grid>

                <Grid item xs={12}>
                    <SearchField
                        value={searchField}
                        onChange={e => setSearchField(e.target.value)}
                        fullWidth
                        margin="dense"
                        variant="filled"
                    />
                </Grid>

                <Grid item xs={5}>
                    <SearchBy
                        options={searchByOptions}
                        selectedOptionId={searchBy}
                        onOptionChange={setSearchBy}
                    />
                </Grid>

                <Grid item>
                    <ColorButton
                        fontcolor="#fff"
                        backgroundcolor={pink[400]}
                        type="submit"
                        size="large"
                    >
                        Search
                    </ColorButton>
                </Grid>
            </Grid>
        </form>
    );
};

SearchForm.propTypes = {
    onSearch: PropTypes.func.isRequired
};
