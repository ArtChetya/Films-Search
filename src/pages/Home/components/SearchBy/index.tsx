import grey from '@material-ui/core/colors/grey';
import pink from '@material-ui/core/colors/pink';
import React, { FunctionComponent } from 'react';
import { IFilterConfig, IOption } from '../../types';
import { FilterBy } from '../FilterBy';

interface ISearchByProps {
    options: IOption[];
    selectedOptionId: string;
    onOptionChange: (id: string) => void;
}

export const SearchBy: FunctionComponent<ISearchByProps> = props => {
    const searchByConfig: IFilterConfig = {
        title: 'Search By',
        titleVariant: 'body1',
        activeColor: '#fff',
        defaultColor: '#fff',
        activeBgColor: pink[400],
        defaultBgColor: grey[700],
        titleTextTransform: 'uppercase',
        buttonTextTransform: 'uppercase',
        buttonVariant: 'outlined'
    };

    return <FilterBy {...props} config={searchByConfig} />;
};
