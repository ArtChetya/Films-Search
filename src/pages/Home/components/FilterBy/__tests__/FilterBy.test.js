import { shallow, mount } from 'enzyme/build';
import React from 'react';
import { FilterBy } from '..';
import pink from '@material-ui/core/colors/pink';
import grey from '@material-ui/core/colors/grey';
import { ColorButton } from 'components';

describe('FilterBy component', () => {
    let config;
    let options;
    let selectedOptionId;
    let onOptionChange;

    beforeAll(() => {
        config = {
            title: 'Search By',
            titleVariant: 'body1',
            activeColor: '#fff',
            defaultColor: '#fff',
            activeBgColor: pink[400],
            defaultBgColor: grey[700],
            titleTextTransform: 'uppercase',
            buttonTextTransform: 'uppercase',
            buttonVariant: 'outlined'
        };

        options = [
            { id: 'title', label: 'Title' },
            { id: 'genres', label: 'Genre' }
        ];

        selectedOptionId = 'title';

        onOptionChange = jest.fn();
    });

    it('should be rendered correctly', () => {
        const component = shallow(
            <FilterBy
                config={config}
                options={options}
                selectedOptionId={selectedOptionId}
                onOptionChange={onOptionChange}
            />
        );
        expect(component).toMatchSnapshot();
    });

    it('should call onOptionChange when option is changed', () => {
        const selectedId = options[options.length - 1].id;
        const component = mount(
            <FilterBy
                config={config}
                options={options}
                selectedOptionId={selectedOptionId}
                onOptionChange={onOptionChange}
            />
        );
        component
            .find(ColorButton)
            .last()
            .simulate('click');

        expect(onOptionChange).toHaveBeenCalledWith(selectedId);
    });
});
