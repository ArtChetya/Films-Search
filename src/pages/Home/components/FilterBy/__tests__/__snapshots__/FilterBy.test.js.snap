// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBy component should be rendered correctly 1`] = `
<WithStyles(Grid)
  alignItems="center"
  container={true}
  spacing={8}
>
  <WithStyles(Grid)
    item={true}
  >
    <TransformText
      fontcolor="#fff"
      transform="uppercase"
      variant="body1"
    >
      Search By
    </TransformText>
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
    key="title"
  >
    <ColorButton
      backgroundcolor="#ec407a"
      onClick={[Function]}
      variant="outlined"
    >
      <TransformText
        fontcolor="#fff"
        transform="uppercase"
        variant="body1"
      >
        Title
      </TransformText>
    </ColorButton>
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
    key="genres"
  >
    <ColorButton
      backgroundcolor="#616161"
      onClick={[Function]}
      variant="outlined"
    >
      <TransformText
        fontcolor="#fff"
        transform="uppercase"
        variant="body1"
      >
        Genre
      </TransformText>
    </ColorButton>
  </WithStyles(Grid)>
</WithStyles(Grid)>
`;
